### DISCUSSION TITLE
Missing py.typed

### DISCUSSION THREAD

**Discussion by andrewmwhite** [encode/starlette#1959] [Potential Issue] [answered]
The `py.typed` file is missing since 0.13.2 from the tarballs on pypi: 

```
$ for fn in *.tar.gz; do echo -ne "$fn: "; tar -tf $fn | grep py.typed || echo; done
starlette-0.12.9.tar.gz: starlette-0.12.9/starlette/py.typed
starlette-0.13.0.tar.gz: starlette-0.13.0/starlette/py.typed
starlette-0.13.2.tar.gz: starlette-0.13.2/starlette/py.typed
starlette-0.13.3.tar.gz: 
starlette-0.13.4.tar.gz: 
starlette-0.13.8.tar.gz: 
starlette-0.14.2.tar.gz: 
starlette-0.15.0.tar.gz: 
starlette-0.20.4.tar.gz: 
```

I'm not clear on why... running the build script in the git repo for tags 0.20.4 and 0.22.0 shows the file there:

```
$ for fn in dist/starlette*.tar.gz; do echo -ne "$fn: "; tar -tf $fn | grep py.typed || echo; done
dist/starlette-0.20.4.tar.gz: starlette-0.20.4/starlette/py.typed
dist/starlette-0.22.0.tar.gz: starlette-0.22.0/starlette/py.typed
```  

This causes issues like the following with mypy when we consume this code (assuming you're not ignoring imports):
```error: Skipping analyzing 'starlette.middleware.base': found module but no type hints or library stubs```

No time now at the moment to dig further.

**Comment by Kludex** [comment, marked as answer]
When I download them on PyPI I can see the `py.typed` there...

Ref.: https://pypi.org/project/starlette/#files

**Reply by andrewmwhite** [reply]
Which specific file are you looking at?

**Reply by Kludex** [reply]
Not sure if I get the question... If I download the files on the link above, I can see the `py.typed`. :thinking:

**Reply by andrewmwhite** [reply]
I was asking which specific version you were looking at and whether you were looking at the tarball or the wheel.

I pulled the latest three tarballs  — it looks like the file is missing from 20.4 but present in 21.0 and 22.0? I’m not sure why.

(Apologies for the screenshots; I only have my tablet with me at the moment.)
![011B868C-1373-41EF-9790-7BF1B701F2DA](https://user-images.githubusercontent.com/1322614/204850495-00e13fff-126e-45e8-9c34-f5ceea7bd789.png)
![DB5E7C5A-0146-4FDA-A96B-DAF1B51F2785](https://user-images.githubusercontent.com/1322614/204850510-8a9b6386-a8b4-4383-8350-097179054206.png)
![145E6E55-FA92-45F5-8047-AED614141562](https://user-images.githubusercontent.com/1322614/204850514-c32b6865-aab2-42c0-9903-375f5969a9f1.png)

**Reply by Kludex** [reply]
Interesting. Well... I think it was on 0.21.0 that we changed our build system to [hatch](https://hatch.pypa.io/latest/). :thinking:

I don't know what happened after 0.13.2.

**Reply by andrewmwhite** [reply]
Bumped our stuff internally to 0.21.0
